src/api/routes/conference.ts

```diff
@@ -189,178 +190,10 @@ export const conference = new Elysia()
-			const singleParticipantsByRole = roles.map((role) => ({
-				role: role.name,
-				fontAwesomeIcon: role.fontAwesomeIcon || undefined,
-				total: role.singleParticipant.length,
-				applied: role.singleParticipant.filter((sp) => sp.applied).length,
-				notApplied: role.singleParticipant.filter((sp) => !sp.applied).length
-			}));
-			const totalApplied = delegationMembersApplied + singleParticipantsApplied;
-			const totalNotApplied =
-				delegations.reduce((acc, d) => acc + d._count.members, 0) +
-				singleParticipants.length -
-				totalApplied;
-
-			const registrationStatistics = {
-				total: totalApplied + totalNotApplied,
-				notApplied: totalNotApplied,
-				applied: totalApplied,
-				delegations: {
-					total: delegationsNotApplied + delegationsApplied,
-					notApplied: delegationsNotApplied,
-					applied: delegationsApplied,
-					withSupervisor: delegationsWithSupervisor
-				},
-				delegationMembers: {
-					total: delegationMembersNotApplied + delegationMembersApplied,
-					notApplied: delegationMembersNotApplied,
-					applied: delegationMembersApplied
-				},
-				singleParticipants: {
-					total: singleParticipantsNotApplied + singleParticipantsApplied,
-					notApplied: singleParticipantsNotApplied,
-					applied: singleParticipantsApplied,
-					byRole: singleParticipantsByRole
-				},
-				supervisors: supervisors.length
-			};
-
-			// Age statistics
-
-			const usersBirthdays = await db.user.findMany({
-				where: {
-					OR: [
-						{ singleParticipant: { some: { conferenceId, applied: true } } },
-						{
-							delegationMemberships: {
-								some: {
-									conferenceId,
-									delegation: {
-										applied: true
-									}
-								}
-							}
-						}
-					]
-				},
-				select: {
-					birthday: true
-				}
-			});
-
-			const agesAtConference = usersBirthdays
-				.filter((u) => u.birthday)
-				.map((u) => {
-					const referenceDate = conference.end
-						? conference.end
-						: conference.start
-							? conference.start
-							: new Date();
-					const age = referenceDate.getTime() - u.birthday!.getTime();
-					return Math.floor(age / (1000 * 60 * 60 * 24 * 365));
-				});
-
-			const averageAge =
-				agesAtConference && agesAtConference.length !== 0
-					? parseFloat(
-							(
-								agesAtConference.reduce((acc, age) => acc + age, 0) / agesAtConference.length
-							).toFixed(1)
-						)
-					: undefined;
-			const ageDistribution: Record<string, number> = {};
-			for (let i = 10; i <= 25; i++) {
-				if (!agesAtConference.includes(i)) continue;
-				ageDistribution[i.toString()] = agesAtConference.filter((age) => age === i).length;
-			}
-
-			const ageStatistics = {
-				average: averageAge,
-				distribution: ageDistribution
-			};
+			const { countdowns, registrationStatistics, ageStatistics } = await conferenceStats(
+				db as unknown as PrismaClient, // TODO @m1212e how do I type the fn param properly, so that I don't need to cast here?
```

Comment:
@m1212e how do I type this properly?

Reply:
I'd just change the type of the extended prisma client. We don't really care for the whole prisma extension type stuff so we just overwrite it. I pushed the changes, feel free to take a look if you want, otherwise this looks good to me, feel free to merge!

Reply:
Perfect, thanks!